### PR TITLE
Add anchor for ShapeComponent constructor

### PR DESCRIPTION
**Pull request by wenxiangjiang**
# Description

Add `anchor` for `ShapeComponent` constructor, so we can easily set the anchor.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by spydon** [comment]
I don't think this will work since the shape is defined from the center, but I'm on my phone now so I can't test it. Have you tried it?

**Comment by wenxiangjiang** [comment]
@spydon Yes, it works well in my project. I added a rectangular box at the top of the game.

Before:

```
var s = ShapeComponent(
  Rectangle(position: Vector2.zero(), size: Vector2(size.x, 80)),
  Paint()..color = Color(0xFF0E1167),
);
s.anchor = Anchor.topLeft;
add(s);
```

Now:

```
var s = ShapeComponent(
  Rectangle(position: Vector2.zero(), size: Vector2(size.x, 80)),
  Paint()..color = Color(0xFF0E1167),
  anchor: Anchor.topLeft,
);
add(s);
```

**Comment by spydon** [comment]
That's great, I'll check tomorrow when I'm not in the park.
Thanks for the contribution! :)

**Review by spydon — APPROVED** [review]

File: packages/flame/lib/src/components/shape_component.dart [hunk 1]
```diff
@@ -12,12 +12,13 @@ class ShapeComponent extends PositionComponent {
   ShapeComponent(
     this.shape,
     this.shapePaint, {
+    Anchor? anchor,
```
Comment (spydon) [inline packages/flame/lib/src/components/shape_component.dart:15]:
```suggestion
    Anchor anchor = Anchor.center,
```

**Review by spydon — COMMENTED** [review]
Tried it and it works well, can you please set the anchor's default like in the suggestion instead of allowing null?

**Review by spydon — APPROVED** [review]

**Pull request merged**